From 23de7a521e3f7553c432d2e833e0abf67898ac29 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Fri, 22 Jun 2018 19:34:36 +0200
Subject: [PATCH] joyent/node-zkstream#33 `Client#isConnected()` throws
 TypeError Reviewed by: Alex Wilson <alex.wilson@joyent.com> Approved by: Alex
 Wilson <alex.wilson@joyent.com>

---
 lib/client.js      | 4 ++--
 package.json       | 2 +-
 test/basic.test.js | 2 ++
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/client.js b/lib/client.js
index bbf0a9d..bb76ef7 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 module.exports = { Client: ZKClient };
@@ -213,7 +213,7 @@ ZKClient.prototype._newSession = function () {
 };
 
 ZKClient.prototype.isConnected = function () {
-	var conn = this.getConnection();
+	var conn = this.currentConnection();
 	return (conn !== undefined && conn.isInState('connected'));
 };
 
diff --git a/package.json b/package.json
index bc72a66..2b8146c 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "zkstream",
-  "version": "0.11.0",
+  "version": "0.11.1",
   "description": "",
   "repository": {
     "type": "git",
diff --git a/test/basic.test.js b/test/basic.test.js
index af4fad0..9b1a265 100644
--- a/test/basic.test.js
+++ b/test/basic.test.js
@@ -102,6 +102,7 @@ mod_tape.test('simple connect and ping', function (t) {
 	});
 
 	zkc.on('connect', function () {
+		t.ok(zkc.isConnected());
 		zkc.ping(function (err) {
 			t.error(err);
 			pinged = true;
@@ -121,6 +122,7 @@ mod_tape.test('double ping', function (t) {
 
 	zkc.on('close', function () {
 		t.strictEqual(pinged, 2);
+		t.notOk(zkc.isConnected());
 		t.end();
 	});
 
-- 
2.21.0

